vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java

```diff
@@ -556,6 +556,7 @@ protected boolean valueEquals(T value1, T value2) {
     private void updateEnabled(RadioButton<T> button) {
         boolean disabled = isDisabledBoolean()
                 || !getItemEnabledProvider().test(button.getItem());
+        button.setEnabled(!disabled);
```

Comment:
This might have security implications. I think the button's enabled status should come from the group status, like so
```java
boolean disabled = !this.isEnabled() ||   !getItemEnabledProvider().test(button.getItem());
``` 
I also think the code below this line should be place on the method `onEnabledStateChanged` in the `RadioButton` class.